lib/game_stats.rb

```diff
@@ -9,6 +9,54 @@ def initialize(file_path, object)
     @games = csv_data('./data/games_truncated.csv', Game)
   end
 
+  def percentage_ties
+    ties = @games.count do |game|
+      game.away_goals == game.home_goals
+    end
+    (100 * ties.fdiv(@games.length)).round(2)
+  end
+
+  def count_of_games_by_season
+    @games.reduce(Hash.new(0)) do |games_by_season, game|
+      games_by_season[game.season] += 1
+      games_by_season
+    end
+  end
+
+  def percentage_home_wins
+    home_wins = @games.find_all {|game| game.away_goals < game.home_goals}
+    sum = (home_wins.length).to_f / (@games.length).to_f
+    (100 * sum).round(2)
+  end
+
+  def percentage_visitor_wins
+    vistor_wins = @games.find_all {|game| game.away_goals > game.home_goals}
+    sum = (vistor_wins.length).to_f / (@games.length).to_f
+    (100 * sum).round(2)
+  end
+
+  def average_goals_per_game
+    all_goals = @games.sum {|game| game.away_goals + game.home_goals}
+    sum = all_goals.to_f / @games.length
+    sum.round(2)
+  end
+
+  def average_goals_by_season
+    goals_per_season = {}
+    @games.each do |game|
+      if goals_per_season[game.season] == nil
+        goals_per_season[game.season] = game.away_goals + game.home_goals
+      else
+        goals_per_season[game.season] += game.away_goals + game.home_goals
+      end
```

Comment:
Future opportunity to refactor into a reduce enumerable. 
`@games.reduce(Hash.new(0)) do |game|
    goals_per_season[game.season] += game.away_goals + game.home_goals
    goals_per_season
end`